### PR TITLE
Issue: Early withdrawal claims

### PR DESCRIPTION
**Pull request by TheTaconator**
Fix of bug that permits withdrawal claims before the first withdrawal period

### CODE REVIEW

**Comment by abitmore** [comment]
I guess this fix requires a hard fork.

**Comment by TheTaconator** [comment]
Thank you for the example case.  The fix and tests have been updated to consider a new fork time.  The fork time has been set to February 1, 2017.

**Comment by abitmore** [comment]
@vikramrajkumar please review.

**Comment by vikramrajkumar** [comment]
@TheTaconator @abitmore This is excellent work and I have not missed this. Just looking into some other logistics of Graphene vs. BitShares 2 and I will come back to this. Thanks for the great work.

File: libraries/app/database_api.cpp [hunk 1]
```diff
@@ -1018,7 +1018,7 @@ market_ticker database_api_impl::get_ticker( const string& base, const string& q
    result.lowest_ask = 0;
    result.highest_bid = 0;
 
-   auto price_to_real = [&]( const share_type a, int p ) { return double( a.value ) / pow( 10, p ); };
+//   auto price_to_real = [&]( const share_type a, int p ) { return double( a.value ) / pow( 10, p ); };
```
Comment (pmconrad) [inline libraries/app/database_api.cpp]: This change looks unrelated to the issue.

File: libraries/chain/include/graphene/chain/withdraw_permission_object.hpp [hunk 2]
```diff
@@ -69,6 +110,23 @@ namespace graphene { namespace chain {
               ? withdrawal_limit.amount - claimed_this_period
               : 0, withdrawal_limit.asset_id );
         }
+
+       /***
+        * Get a description of the current withdrawal period
+        * @param current_time   Current time
+        * @return A description of the current period
+        */
+       withdrawal_period_descriptor current_period(fc::time_point_sec current_time) const {
```
Comment (pmconrad) [inline libraries/chain/include/graphene/chain/withdraw_permission_object.hpp]: AFAICS the withdrawal_period_descriptor and this method are only used in unit tests. IMO they shouldn't become part of the main code in that case.

File: libraries/app/database_api.cpp (hunk 1 above)
Comment (TheTaconator) [inline libraries/app/database_api.cpp]: That is correct. I noticed that it was an unused variable. Do you prefer that I remove it from the commit?

File: libraries/chain/include/graphene/chain/withdraw_permission_object.hpp (hunk 2 above)
Comment (TheTaconator) [inline libraries/chain/include/graphene/chain/withdraw_permission_object.hpp]: Fair point, although I have this object being used for code (that is not yet committed) which is related to informing a user/client software about what can still be withdrawn during any particular period.

Do you suggest that I:
(1) temporarily move this code out of the main code, and then later
(2) move this class into the main code when committing my future code?

Comment (pmconrad) [inline libraries/chain/include/graphene/chain/withdraw_permission_object.hpp]: Yes please.

File: libraries/app/database_api.cpp (hunk 1 above)
Comment (pmconrad) [inline libraries/app/database_api.cpp]: Please rebase on develop and see if this change remains. In that case I'd be OK with it, but instead of commenting out the line just remove it completely.

**Comment by pmconrad** [comment]
@TheTaconator do you think you can add the requested modifications within the next couple of days?

**Comment by abitmore** [comment]
Also "this branch has conflicts that must be resolved". Need to rebase.

**Comment by TheTaconator** [comment]
@pmconrad apologies, but due to some other deadlines I will not be able to make the changes in time. I might need another 7 days. I have no objections about proceeding without this fix for now.

**Comment by pmconrad** [comment]
Closing here, continued in #386
